### PR TITLE
feat(cloudzap): set default slog logger to write to zap

### PR DESCRIPTION
**Pull request by Edholm**
A new package is added: cloudslog that is very simple at the moment. It
just contains a func to create a new slog logger from a zap logger.

This change also replaces the default log/slog logger to a one that
instead logs to our default zap logger. This is useful if you use libs
or sdks that do not rely on cloudrunner and/or zap.

The following code produces these logs:
```go
cloudrunner.Logger(ctx).Info("hello from cloudrunner.Logger(ctx)", zap.String("string", "str"), zap.Int("int", 32))
slog.Info("hello from slog.Info", "string", "str", "int", 32)
log.Printf("hello from log.Printf string=%s, int=%d", "str", 32)

```

## Dev logs
image

## Prod logs
image

### CODE REVIEW

File: cloudzap/logger.go [hunk 1]
```diff
@@ -24,10 +24,14 @@ func NewLogger(config LoggerConfig) (*zap.Logger, error) {
...
 		zapConfig.Level = zap.NewAtomicLevelAt(config.Level)
-		return zapConfig.Build(
+		logger, err := zapConfig.Build(
 			zap.AddCaller(),
 			zap.AddStacktrace(zap.FatalLevel), // add stacktraces manually where needed
 		)
+		if err == nil {
+			setSlogDefault(logger.Core())
+		}
+		return logger, err
```
Comment (radhus) [inline cloudzap/logger.go:34]: Understand this is matching the previous change (returning both values), but would probably prefer the more conventional err check 😄 
```suggestion
		if err != nil {
			return nil, fmt.Errorf("init logger: %w", err)
		}
		setSlogDefault(logger.Core())
		return logger, nil
```

File: cloudzap/logger.go [hunk 2]
```diff
@@ -24,10 +24,14 @@ func NewLogger(config LoggerConfig) (*zap.Logger, error) {
 		zapConfig := zap.NewDevelopmentConfig()
 		zapConfig.EncoderConfig.EncodeLevel = zapcore.LowercaseColorLevelEncoder
 		zapConfig.Level = zap.NewAtomicLevelAt(config.Level)
-		return zapConfig.Build(
+		logger, err := zapConfig.Build(
 			zap.AddCaller(),
 			zap.AddStacktrace(zap.FatalLevel), // add stacktraces manually where needed
 		)
+		if err == nil {
+			setSlogDefault(logger.Core())
```
Comment (odsod) [inline cloudzap/logger.go:32]: IMO this function is a constructor and shouldn't modify global state - let's do this one level up closer to the `cloudrunner.Run` function?

**Review by radhus — APPROVED** [review]
Great to have compat! ⭐

**Review by odsod — CHANGES_REQUESTED** [review]
See comment on global state

File: cloudslog/logger.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+import (
+	"log/slog"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/exp/zapslog"
+)
+
+// NewFromZap returns a slog logger in which the underlying handler writes to the given zap logger.
+func NewFromZap(zl *zap.Logger) *slog.Logger {
```
Comment (odsod) [inline cloudslog/logger.go:11]: Maybe we should avoid exposing this altogether actually, so we can yeet it without regret? Just put it as a private function in `run.go`?

Comment (Edholm) [reply cloudslog/logger.go:11]: That is a good point, makes a possible future deprecation of zap a lot easier :)

**Review by odsod — APPROVED** [review]

File: cloudslog/logger.go (hunk 3 above)
Comment (Edholm) [reply cloudslog/logger.go]: Thanks, I'll wait for @radhus to also have a look :)

**Review by radhus — APPROVED** [review]
🚀

**Pull request merged**